src/main/java/Duke.java

```diff
@@ -1,10 +1,151 @@
+    }
+
+    private static void handleCommand(String line) {
+        String description;
+        String date;
+        dones.add(false);
+        if (line.startsWith("todo")) {
+            types.add("T");
+            description = line.substring(5);
+            Todo todo = new Todo(description);
+            descriptions.add(description);
+            dates.add("");
+            System.out.println("       " + todo.toString(false));
+        } else if (line.startsWith("deadline")) {
+            types.add("D");
+            int separator = line.indexOf("/by");
+            description = line.substring(9, separator - 1);
+            date = line.substring(separator + 4);
+            Deadline deadline = new Deadline(description, date);
+            descriptions.add(description);
+            dates.add(date);
+            System.out.println("       " + deadline.toString(false));
+        } else if (line.startsWith("event")) {
+            types.add("E");
+            int separator = line.indexOf("/at");
+            description = line.substring(6, separator - 1);
+            date = line.substring(separator + 4);
+            Event event = new Event(description, date);
+            descriptions.add(description);
+            dates.add(date);
+            System.out.println("       " + event.toString(false));
+        } else {
+            types.add("unknown");
+        }
+    }
+
+    private static void showList() {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Here are the tasks in your list:");
+        for (int i = 0; i < descriptions.size(); i++) {
+            String description = descriptions.get(i);
+            String date = dates.get(i);
+            String type = types.get(i);
+            boolean isDone = dones.get(i);
+            System.out.printf("     %d.", i + 1);
+            switch (type) {
+                case "T":
+                    Todo todo = new Todo(description);
+                    System.out.println(todo.toString(isDone));
+                    break;
+                case "D":
+                    Deadline deadline = new Deadline(description, date);
+                    System.out.println(deadline.toString(isDone));
+                    break;
+                case "E":
+                    Event event = new Event(description, date);
+                    System.out.println(event.toString(isDone));
+                    break;
+                default:
+                    System.out.println("       Unknown Type");
+            }
+        }
+        System.out.println("    ____________________________________________________________");
+    }
+
+    private static void markAsDone(int number) {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Nice! I've marked this task as done:");
+        int index = number - 1;
```

Comment:
Consider naming the variable as something that may be more easily understood at one glance, e.g. taskNumber, taskIndex

Reply:
Thank you for pointing it out, I will work on it!